pkg/pool-weighted/contracts/managed/vendor/BasePool.sol

```diff
@@ -332,8 +329,6 @@ abstract contract BasePool is
         uint256 protocolSwapFeePercentage,
         bytes memory userData
     ) external override returns (uint256 bptOut, uint256[] memory amountsIn) {
-        InputHelpers.ensureInputLengthMatch(balances.length, _getTotalTokens());
```

Comment:
This is safe to do because we also don't check the length in eg. `onJoinPool`. `_onJoin` and `_onExit` **must** implement this check.

Reply:
However we have must stronger trust in the caller in those cases. If the Vault has fallen out of sync with the Pool then the Pool has acted improperly. Users can provide arbitrary data however.